models/users.js

```diff
@@ -119,29 +119,35 @@ const getSuggestedUsers = async (skill) => {
 
 /**
  * Fetches the data about our users
- * @param query { Object }: Filter for users
+ * @param query { search, next, prev, size, page }: Filter for users
  * @return {Promise<userModel|Array>}
  */
 const fetchUsers = async (query) => {
-  const appendUsernamePrefixQuery = (dbQuery) => {
-    return dbQuery
-      .orderBy("username")
-      .startAt(query.search.toLowerCase().trim())
-      .endAt(query.search.toLowerCase().trim() + "\uf8ff");
-  };
   try {
     // INFO: default user size set to 100
     // INFO: https://github.com/Real-Dev-Squad/website-backend/pull/873#discussion_r1064229932
     const size = parseInt(query.size) || 100;
-    const page = size * (parseInt(query.page) || 0);
-
-    let dbQuery = userModel.limit(size).offset(page);
+    const doc = (query.next || query.prev) && (await userModel.doc(query.next || query.prev).get());
+    let dbQuery = query.prev ? userModel.limitToLast(size) : userModel.limit(size);
+    dbQuery = dbQuery.orderBy("username");
     if (Object.keys(query).length) {
       if (query.search) {
-        dbQuery = appendUsernamePrefixQuery(dbQuery);
+        dbQuery = dbQuery
+          .startAt(query.search.toLowerCase().trim())
+          .endAt(query.search.toLowerCase().trim() + "\uf8ff");
+      } else if (query.page) {
+        const offsetValue = size * parseInt(query.page);
+        dbQuery = dbQuery.offset(offsetValue);
+      } else if (query.next) {
+        dbQuery = dbQuery.startAfter(doc);
+      } else if (query.prev) {
+        dbQuery = dbQuery.endBefore(doc);
```

Comment:
Will pagination be applied for search results also?

Reply:
yes, but it wouldn't add in the next and prev links, guess i need to make a change then

Reply:
How util function handling search paginations

Reply:
sorry got your point, if we have search param it will skip page/next/prev and everything and not be paginated, guess i forgot my point using only `if` in the first place and changed it to `if-else`

Reply:
Util function doesn't handle pagiation, it was created to create `links` based on the request params